jerry-core/ecma/operations/ecma-shared-arraybuffer-object.c

```diff
@@ -166,15 +168,34 @@ ecma_op_create_shared_arraybuffer_object (const ecma_value_t *arguments_list_p,
  * @return true - if value is a SharedArrayBuffer object
  *         false - otherwise
  */
-bool
+extern inline bool JERRY_ATTR_ALWAYS_INLINE
 ecma_is_shared_arraybuffer (ecma_value_t target) /**< the target value */
 {
+#if JERRY_BUILTIN_SHAREDARRAYBUFFER
   return (ecma_is_value_object (target)
           && ecma_object_class_is (ecma_get_object_from_value (target), ECMA_OBJECT_CLASS_SHARED_ARRAY_BUFFER));
+#endif /* JERRY_BUILTIN_SHAREDARRAYBUFFER */
+  JERRY_UNUSED (target);
+  return false;
```

Comment:
This should be:
```c
#if ...
  return (ecma_is...
#else /* !... */
  JERRY_UNUSED (...);
  return false;
#endif
```

The `#else` case is explicitly written out. Otherwise if the shared arraybuffers are enabled the code would look like:
```c
return (...)
JERRY_UNUSED (target)
return false;
```
in which the last two lines are dead-code.